provider/provider.go

```diff
@@ -59,33 +90,46 @@ type Tiler interface {
 	TileFeatures(ctx context.Context, layer string, t Tile, fn func(f *Feature) error) error
 }
 
-type Layerer interface {
-	// Layers returns information about the various layers the provider supports
-	Layers() ([]LayerInfo, error)
+// TilerUnion represents either a Std Tiler or and MVTTiler; only one should be not nil.
+type TilerUnion struct {
+	Std Tiler
+	Mvt MVTTiler
 }
 
-type LayerInfo interface {
-	Name() string
-	GeomType() geom.Geometry
-	SRID() uint64
+func (tu TilerUnion) Layers() ([]LayerInfo, error) {
+	if tu.Std != nil {
+		return tu.Std.Layers()
+	}
+	if tu.Mvt != nil {
+		return tu.Mvt.Layers()
+	}
+	return nil, ErrNilInitFunc
```

Comment:
It could be. It could, also, be a panic. Was not sure which of the three options made the most sense. When with NilInitFunc because the only way one should get here is if someone added a new provider type and forgot to register the provider type in this area of the code, but maybe panic is better in that sense? I went with ErrNilFunc because that's the error you get from the init functions, and it's not panic. And having another error case got a case that should never occur did not seem right. (But not I'm leaning more towards the panic)

Reply:
I don't know if a panic would be too user friendly. I suspect this error will most commonly be encountered when someone fat fingers a provider type, right? For example, putting in `mvt_postgres` instead of `mvt_postgis`. If that's the case, we could report an error to guide the user to resolution.

Reply:
In reply to:
https://github.com/go-spatial/tegola/pull/700#discussion_r453957361


No, that error would be caught by the Config Validation long before it reaches here.

Reply:
> I don't know if a panic would be too user friendly. I suspect this error will most commonly be encountered when someone fat fingers a provider type, right? For example, putting in `mvt_postgres` instead of `mvt_postgis`. If that's the case, we could report an error to guide the user to resolution.

No this would be caught by the config validation code earlier.